std/complex.d

```diff
@@ -142,15 +143,16 @@ struct Complex(T)  if (isFloatingPoint!T)
     }
 
     /// ditto
-    void toString(Char)(scope void delegate(const(Char)[]) sink,
+    void toString(Writer, Char)(scope Writer w,
                         FormatSpec!Char formatSpec) const
```

Comment:
static assert could do better at explaining to the user what wrong. Template contraints actually have worse error messages than no constraints, at least in my experience.